adflow/mphys/mphys_adflow.py

```diff
@@ -1147,11 +1149,17 @@ def __init__(
         restart_failed_analysis=False,  # retry after failed analysis
         err_on_convergence_fail=False,  # raise an analysis error if the solver stalls
         balance_group=None,
+        my_surfs=None,
```

Comment:
As an explanation, the order MPhys was initializing ADflow made it impossible to set custom surface family groups.  This is a bit of a hack for allowing users to add a custom surface family, but I think it might need to be expanded using a dictionary to add multiple custom surfaces.  I will make these changes and add documentation to explain what's happening.

Reply:
> Also, I think other BC routines also have similar issues, its not just the one you fixed. In fact, the others just terminate the code instead of hanging. E.g. https://github.com/mdolab/adflow/blob/main/src/bcdata/BCData.F90#L969 Can we apply the same fix to the others? Are you sure the code does not hang and the processors end up with the correct fail flags when you do the allreduce in python?

I can confirm that the procs end up with the correct fail flags when I do the allreduce in the python layer.  I've tested this functionality with my application on hpc, but also with smaller toy problems locally.  I can look into adding tests for this behavior, but it might be involved to manufacture a toy problem at a small scale where the BC data is only on some procs and not others.